code/go/0chain.net/chaincore/config/config.go

```diff
@@ -135,30 +137,69 @@ const (
 	DeploymentMainNet     = 2
 )
 
+type ConfigReader interface {
+	ReadValue(name string) (interface{}, error)
+}
+
+// This is defined to avoid 'nil pointer reference' to ConfigChain
+// And also to be used for setting the configuration for Test purposes
+type TestConfigReader struct {
+	Fields map[string]interface{}
+	mu     sync.Mutex
+}
+
+func (cr *TestConfigReader) ReadValue(name string) (interface{}, error) {
+	cr.mu.Lock()
+	defer cr.mu.Unlock()
+
+	v, found := cr.Fields[name]
+	if !found {
+		return nil, errors.New("ChainConfig - Read Config - Invalid configuration name")
+	}
+	return v, nil
+}
+
+func (cr *TestConfigReader) WriteValue(name string, val interface{}) {
+	cr.mu.Lock()
+	defer cr.mu.Unlock()
+
+	cr.Fields[name] = val
+}
+
 /*Config - all the config options passed from the command line*/
 type Config struct {
 	Host           string
 	Port           int
 	ChainID        string
 	DeploymentMode byte
+	ChainConfig    ConfigReader
 }
 
 /*Configuration of the system */
-var Configuration Config
+var configuration Config
+
+func Configuration() *Config {
+	if configuration.ChainConfig == nil {
+		configuration.ChainConfig = &TestConfigReader{
```

Comment:
I did initialize it explicitly at the time of creating the chain:
https://github.com/0chain/0chain/blob/df022bb2d701af666a428fe0c87a300f598c41f1/code/go/0chain.net/chaincore/chain/entity.go#L418-L419
But I was suspicious we may access it inside a worker before we initialize it. So I added this Dummy ConfigReader and I set it on first access to Configuration object. Otherwise we have to check 
`if config.Configuration.ChainConfig != nil {
  config.Configuration.ChainConfig.ReadValue(...)
}` which is not convenient to use.